### PR TITLE
jwt and pangea token support

### PR DESCRIPTION
**Pull request by pangea-andrest**

### CODE REVIEW

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 1]
```diff
@@ -259,3 +262,60 @@ class UpdateRequest(APIRequestModel):
...
+class JWTGetRequest(APIRequestModel):
+    id: str
+    version: Optional[str] = None
+
+
+class JWKHeader(PangeaResponseResult):
+    alg: str
+    kid: Optional[str] = None
+    kty: str
+    use: Optional[str] = None
```
Comment (dbenders) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:276]: I think we're not sending this anymore. @dulanowiczpangea ?

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 2]
```diff
@@ -259,3 +262,60 @@ class UpdateRequest(APIRequestModel):
...
 class UpdateResult(APIRequestModel):
     id: str
+
+
+class JWTGetRequest(APIRequestModel):
+    id: str
+    version: Optional[str] = None
+
+
+class JWKHeader(PangeaResponseResult):
```
Comment (dbenders) [inline packages/pangea-sdk/pangea/services/vault/models/common.py:272]: I'm not sure if we need to interpret this, or just leave it as a dict... the user probably has a library to manipulate them

File: packages/pangea-sdk/pangea/services/vault/vault.py [hunk 3]
```diff
@@ -199,14 +206,51 @@ def secret_store(
...
+    def pangea_token_store(
+        self,
+        pangea_token: str,
+        name: Optional[str] = None,
+        folder: Optional[str] = None,
+        metadata: Optional[Metadata] = None,
+        tags: Optional[Tags] = None,
+        rotation_policy: Optional[str] = None,
+        auto_rotate: Optional[bool] = None,
+        retain_previous_version: Optional[bool] = None,
```
Comment (dbenders) [inline packages/pangea-sdk/pangea/services/vault/vault.py:218]: not needed for pangea tokens (the previous versions are in fact deleted from Pangea)

File: packages/pangea-sdk/pangea/services/vault/vault.py [hunk 4]
```diff
@@ -199,14 +206,51 @@ def secret_store(
...
+            expiration=expiration,
+        )
+        response = self.request.post("secret/store", data=input.json(exclude_none=True))
+        if response.raw_result is not None:
+            response.result = SecretStoreResult(**response.raw_result)
+        return response
+
     # Rotate endpoint
-    def secret_rotate(self, id: str, secret: str) -> PangeaResponse[SecretRotateResult]:
+    def secret_rotate(self, id: str, secret: Optional[str] = None) -> PangeaResponse[SecretRotateResult]:
```
Comment (dbenders) [inline packages/pangea-sdk/pangea/services/vault/vault.py:239]: for rotating secrets, the value is mandatory

File: packages/pangea-sdk/pangea/services/vault/vault.py [hunk 5]
```diff
@@ -388,3 +436,25 @@ def verify(
...
+
+    def jwt_sign(self, id: str, payload: str) -> PangeaResponse[JWTSignResult]:
+        input = JWTSignRequest(id=id, payload=payload)
+        response = self.request.post("key/sign/jwt", data=input.json(exclude_none=True))
+        if response.raw_result is not None:
+            response.result = JWTSignResult(**response.raw_result)
+        return response
+
+    # Get endpoint
+    def jwt_get(self, id: str, version: Optional[str] = None) -> PangeaResponse[JWTGetResult]:
```
Comment (dbenders) [inline packages/pangea-sdk/pangea/services/vault/vault.py:455]: jwk_get?

File: packages/pangea-sdk/pangea/services/vault/models/common.py [hunk 6]
```diff
@@ -259,3 +262,60 @@ class UpdateRequest(APIRequestModel):
 
 class UpdateResult(APIRequestModel):
     id: str
+
+
+class JWTGetRequest(APIRequestModel):
```
[the remark this answers is not in the document]
Comment (pangea-andrest) [reply packages/pangea-sdk/pangea/services/vault/models/common.py:267]: I'll use JWK

**Review by dbenders — APPROVED** [review]

**Pull request merged**